warmup/summary_statistics.py

```diff
@@ -57,21 +65,54 @@ def collect_summary_statistics(data_dictionaries, delta, steady_state):
                 # steady state. However, the last segment may be equivalent to
                 # its adjacent segments, so we first need to know which segments
                 # are steady-state segments.
+                if data_dictionaries[machine]['classifications'][key][p_exec] == 'no steady state':
+                    continue
+                # Capture the last steady state segment for bootstrapping.
+                segment_data = list()
+                if data_dictionaries[machine]['changepoints'][key][p_exec]:
+                    start = data_dictionaries[machine]['changepoints'][key][p_exec][-1]
+                else:
+                    start = 0  # No changepoints in this pexec.
+                end = len(data_dictionaries[machine]['wallclock_times'][key][p_exec])
+                for segment_index in xrange(start, end):
+                    if segment_index in data_dictionaries[machine]['all_outliers'][key][p_exec]:
+                        continue
+                    segment_data.append(data_dictionaries[machine]['wallclock_times'][key][p_exec][segment_index])
+                segments_for_bootstrap_this_pexec.append(segment_data)
+
                 first_steady_segment = len(data_dictionaries[machine]['changepoint_means'][key][p_exec]) - 1
                 num_steady_segments = 1
                 last_segment_mean = data_dictionaries[machine]['changepoint_means'][key][p_exec][-1]
                 last_segment_var = data_dictionaries[machine]['changepoint_vars'][key][p_exec][-1]
                 lower_bound = min(last_segment_mean - last_segment_var, last_segment_mean - delta)
                 upper_bound = max(last_segment_mean + last_segment_var, last_segment_mean + delta)
+                # This for loop deals with segments that are equivalent to the
+                # final, steady state segment.
                 for index in xrange(len(data_dictionaries[machine]['changepoint_means'][key][p_exec]) - 2, -1, -1):
                     current_segment_mean = data_dictionaries[machine]['changepoint_means'][key][p_exec][index]
                     current_segment_var = data_dictionaries[machine]['changepoint_vars'][key][p_exec][index]
                     if (current_segment_mean + current_segment_var >= lower_bound and
                             current_segment_mean - current_segment_var<= upper_bound):
+                        # Extract this segment from the wallclock data for bootstrapping.
+                        segment_data = list()
+                        if index == 0:
+                            start = 0
+                            end = data_dictionaries[machine]['changepoints'][key][p_exec][index] + 1
+                        else:
+                            start = data_dictionaries[machine]['changepoints'][key][p_exec][index - 1] + 1
```

Comment:
Does this deal wit the final segment correctly? There's an implicit changepoint end at the end of the pexec, right?